src/app/authoring-tool/edit-component-constraints/edit-component-constraints.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { TeacherProjectService } from '../../../assets/wise5/services/teacherProjectService';
+import { Constraint } from '../../domain/constraint';
+
+@Component({
+  selector: 'edit-component-constraints',
+  templateUrl: './edit-component-constraints.component.html',
+  styleUrls: ['./edit-component-constraints.component.scss']
+})
+export class EditComponentConstraintsComponent implements OnInit {
```

Comment:
This class has a lot of duplicated code with NodeAdvancedConstraintAuthoringComponent. Can we refactor to reduce duplication?